---
title: "TCR phospho-signalling upon antibody-based stimulation"
author: "Marie Locard-Paulet"
date: '`r date()`'
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = FALSE)

library(venneuler)
library(reshape2)
library(ggplot2)
library(gplots)
library(knitr)
library(corrplot)

mlpcolours <- c("seagreen3", "brown2", "skyblue4", "tan1", "lightseagreen", "hotpink4", "lightpink1", "blue3")

# Functions for the pairwise scatterplots:
panel.cor <- function(x, y, digits=2, prefix="", cex.cor) {
  usr <- par("usr"); on.exit(par(usr))
  par(usr = c(0, 1, 0, 1))
  r = (cor(x[complete.cases(x) & complete.cases(y)], y[complete.cases(y) & complete.cases(x)]))
  txt <- format(c(r, 0.123456789), digits=digits)[1]
  txt <- paste(prefix, txt, sep="")
  if(missing(cex.cor)) cex <- 0.8/strwidth(txt)
  text(0.5, 0.5, txt, cex = cex * abs(r))
}
panel.hist <- function(x, ...) {
  usr <- par("usr"); on.exit(par(usr))
  par(usr = c(usr[1:2], 0, 1.5) )
  h <- hist(x, plot = FALSE)
  breaks <- h$breaks; nB <- length(breaks)
  y <- h$counts; y <- y/max(y)
  rect(breaks[-nB], 0, breaks[-1], y, col = "cyan", ...)
}
```

# Aim of the analysis

The aim of the analysis is to identify phosphorylation sites that are regulated upon antibody-based TCR activation in primary T-cells. A global analysis of our proteomic/phosphoproteomic data sets should allow us to study phospho-signalling resulting from TCR activation in primary cells.

# Sample preparation and input files

## Cell stimulation

Romain Roncagalli performed cell isolation and stimulation 4 times (R1: 20141204, R3: 20150306, R4: 20150713, R5: 20151103) as follows:

Day 0:

1. Purified CD4+T cells (negative selection dynal, invitrogen cat: 114-15D) are activated in culture with 3 ug/ml of coated (O/N 4C or 4H 37C and wash 3X with PBS1X) anti-CD3 (2C11) + soluble anti-CD28 (1ug/ml). 4 ml/well of 0.625 10^6 cells/ml are distributed in 6 well plates. 
Use complete RPMI 
 
Day 2:

1. Harvest cells, wash and count 
2. Resuspend cells at 1 10^6 cells/ml with IL-2 (1/2000), and distribute them: 5ml/well (6 well plates)
 
Day 3:

Add 5 ml of complete medium with IL-2/ well
 
Day 4:

1. Wash cells 2X with 37 C RPMI SF, count  
2. 50 10^6 of cells are resuspended in 4 ml RPMI SF, pelleted and 10 ug of biotinylated anti-CD3 + 10 ug of biotinylated anti-CD4 were added on pellets. (for ab stimulation). 2X 50 10^6 were prepared for each condition.
3. Pellets are resuspended in 100 ul of  37C SF RPMI and incubated for 5 min at 37C
4. 28 ug (100 ul) of purified Streptavidin (SAV) for ab stimulation and incubated for different time points 37C
5. Stimulation was stopped by freezing tube in liquid nitrogen


## Sample preparation for MS analysis

### Peptide samples

After trypsin digestion, 5ug of each peptide sample was injected on the Q-Exactive plus for relative protein quantification.

### phospho-samples 

Carine Froment performed TiO2 enrichments, injected 3 to 4 times 2.5% of them (labelled "TiO2") and performed a phospho-tyrosine IP on the 90% remaining (injected in 3 to 4 technical replicates - corresponds to 22.5% of starting material). Samples were injected onto a Q-Exactive Plus for label-free MS relative quantification. The analysis was performed using Maxquant.


## Input files

The input files are all in the folder `RAW/`:

* `PeptideSamples/`: output protein table from the MaxQuant analysis of all peptide samples prior to TiO2 and phospho-tyrosine enrichment.
* `TiO2/`: output phosphosites tables from the MaxQuant analysis of 10% of the samples after TiO2 and before phospho-tyrosine enrichment. Each biological replicate was searched independently.
* `pYIP/`: output phosphosites tables from the MaxQuant analysis of the samples after TiO2 and phospho-tyrosine enrichment. Each biological replicate was searched independently.


```{r, eval = F}
files <- list.files("RAW/", pattern = ".txt", recursive = T, full.names = T)
files <- files[(grepl("TiO2", files) | grepl("pYIP", files)) & !grepl("peptides", files)]
lf <- lapply(files, read.delim, stringsAsFactors = F, sep = "\t", header = T)
names(lf) <- gsub("RAW//", "", files, fixed = T)
```

--------------------------------------------------------------------------------

# Multisites parsing

MaxQuant returns a table with one row per phosphorylation site. It returns the quantification values for the multiply phosphorylated site but does not map them to the corresponding pairs (or triplets). I use a script to identify the quantification values comming from the multiply phosphorylated peptides and match them.

I keep only the sites with a PEP value <= 0.01.

```{r, eval = F}
source("RFunctions/MultiPhosphoParsingMQ_20180817.R")

for (i in seq_along(lf)) {
  tab <- lf[[i]]
  tab <- tab[!grepl("CON_", tab$Protein) & !grepl("REV_", tab$Protein),]
  tab <- tab[(tab$PEP <= 0.01),]
  tab <- MultiPhosphoParsingMQ(tab)
  sitespos <- sapply(as.character(tab$phosphoSites), function(x) {
    strsplit(x, "_", fixed = T)[[1]][2]
  })
  lf[[i]] <- tab
}
names(lf) <- sapply(names(lf), function(x) {
  substr(x, start = nchar("pYIP/Phospho (STY)Sites_ProFI-S4_p"), stop = nchar("pYIP/Phospho (STY)Sites_ProFI-S4_pYIP_R1"))
})
```

I correct some issues with protein IDs:

```{r, eval = F}
for (j in seq_along(lf)) {
  ID <- lf[[j]]$phosphoSites
  pos <- sapply(ID, function(x) {
    strsplit(x, "_", fixed = T)[[1]][2]
  })
  # I remove discrepencies in the names of the phosphosites for ambiguous peptides comming from different files:
  
  ## I remove the PrimeXS peptide IDs
  PrimeXS <- ";PrimeXS"
  if ((sum(grepl(PrimeXS, ID, fixed = T)))>0) {
    vec <- lapply(pos[grepl(PrimeXS, ID, fixed = T)], function(x) strsplit(x, ";", fixed = T)[[1]])
    vec <- lapply(1:length(vec), function(x) vec[[x]][1:(length(vec[[x]])-1)])
    vec <- sapply(vec, function(x) {
      paste(x, collapse = ";")
    })
    newID <- gsub(PrimeXS, "", ID[grepl(PrimeXS, ID, fixed = T)])
    newID <- sapply(newID, function(x) {
      strsplit(x, "_")[[1]][1]
    })
    newID <- paste0(newID, "_", vec)
    ID[grepl(PrimeXS, ID, fixed = T)] <- newID
  }
  
  # I reorder the IDs in the alphabetical order of the protein accessions:
  prot <- lapply(ID, function(x) {
    strsplit(x, "_")[[1]][1]
  })
  site <- lapply(ID, function(x) {
    strsplit(x, "_")[[1]][2]
  })
  for (i in seq_along(prot)) {
    if (grepl(";", prot[[i]])){
      veci <- strsplit(prot[[i]], ";", fixed = T)[[1]]
      o <- order(veci)
      if (grepl("+", site[[i]])) {
        veci2 <- strsplit(site[[i]], "+", fixed = T)[[1]]
        aa <- sapply(veci2, function(x) {
          substr(x, 1, 1)
        })
        veci2 <- lapply(veci2, function(x) {
          strsplit(substr(x, 2, nchar(x)), ";", fixed = T)[[1]]
        })
        veci2 <- lapply(seq_along(veci2), function(x) {
          paste0(aa[x], paste(veci2[[x]][o], collapse = ";"))
        })
        veci2 <- paste(unlist(veci2), collapse = "+")
      } else {
        aa <- substr(site[[i]], 1, 1)
        x <- site[[i]]
        veci2 <- strsplit(substr(x, 2, nchar(x)), ";", fixed = T)[[1]]
        veci2 <- paste0(aa, paste(veci2[o], collapse = ";"))
      }
      veci <- paste(veci[o], collapse = ";")
      ID[i] <- paste0(veci, "_", veci2)
    }
  }
  
  # Put together LCK-394 and FYN|YES|SRC-Y420|424|424:
  ID[ID=="P05480;P39688;Q04736_Y424;420;424"] <- "P05480;P06240;P39688;Q04736_Y424;394;420;424"
  ID[ID=="P05480;P06240;Q04736_Y424;394;424"] <- "P05480;P06240;P39688;Q04736_Y424;394;420;424"
  
  lf[[j]]$phosphoSites <- ID
  sitespos <- sapply(ID, function(x) {
    strsplit(x, "_", fixed = T)[[1]][2]
  })
  lf[[j]]$psiteGeneID <- paste0(lf[[j]]$Gene.names, "_", sitespos)
  
  lf[[j]] <- lf[[j]][as.character(lf[[j]]$phosphoSites) != "P06240_Y394",]
}

        
save(lf, file = "RData/01_ParsedTables.RData")

# Save table for troubleshouting:
temp <- merge(lf[[1]], lf[[2]], by = "phosphoSites", all = T)
for (i in 3:length(lf)) {
  temp <- merge(temp, lf[[i]], by = "phosphoSites", all = T)
}
write.table(temp, "ParsedTables.txt", sep = "\t", row.names = F)
```


I create a list with for each phosphorylation site ID its ID with gene name (for figures) and the list of all the mono-phosphorylations it corresponds to.

```{r, eval = F}
vecID <- vector(mode = "list")
vecGeneID <- vector(mode = "list")
lMono <- vector(mode = "list")
for (i in seq_along(lf)) {
  vecID[[i]] <- lf[[i]]$phosphoSites
  vecGeneID[[i]] <- lf[[i]]$psiteGeneID
  vec <- sapply(as.character(lf[[i]]$phosphoSites), function(x) {
    strsplit(x, "_", fixed = T)[[1]][2]
  })
  vecAcc <- sapply(as.character(lf[[i]]$phosphoSites), function(x) {
    strsplit(x, "_", fixed = T)[[1]][1]
  })
  vec <- lapply(seq_along(vec), function(x) {
    paste0(vecAcc[x], "_", strsplit(vec[x], "+", fixed = T)[[1]])
  })
  names(vec) <- vecID[[i]]
  lMono[[i]] <- vec
}
uID <- sort(unique(unlist(vecID)))
mappingID <- unlist(vecID)
vecGeneID <- unlist(vecGeneID)
vecID <- unlist(vecID)

psiteInfo <- vector(mode = "list")
i = 0
for (el in uID) {
  i = i + 1
  vec <- sapply(lMono, function(x) {
    unique(x[names(x) == el])
  })
  vec <- unique(unlist(vec))
  lel <- list("phosphoSites" = el, "GeneID" = unique(vecGeneID[vecID == el]), "MonoSites" = sort(vec))
  psiteInfo[[i]] <- lel
}
names(psiteInfo) <- uID
save(psiteInfo, file = "RData/02_psiteInfo.RData")
```

# Data normalisation using iRTs 

```{r}
load(file = "RData/04_ParsedTablesPSP.RData")
load(file = "RData/03_psiteInfoPhosphoSitePlus.RData")
```

In order to correct for technical variation due to the instrument with spided-in synthetic peptides. I use their intensity to normalise the data.  I keep only the iRTs with a CV < 50%, and take as a reference the median of the runs (iRT signal) in the 4th biological repeat. This is to keep the values close to the ones of the NS in the R4, where there was an issue with the spike and I can't perform normalisation.

```{r}
files <- list.files("RAW/iRTNorm/", pattern = ".txt", recursive = T, full.names = T)
iRT <- lapply(files, read.delim, stringsAsFactors = F, sep = "\t", header = T)
names(iRT) <- gsub("RAW/iRTNorm/peptides_", "", files, fixed = T)
names(iRT) <- gsub(".txt", "", names(iRT), fixed = T)
rm(files)

for (i in seq_along(iRT)) {
  tab <- iRT[[i]]
  tab <- tab[grepl("Biognosys", tab$Proteins),]
  mat <- tab[,grepl("Intensity.", names(tab), fixed = T)]
  mat[mat == 0] <- NA
  # boxplot(log2(mat), las = 2, main = names(iRT)[i])
  cv <- sapply(seq_len(nrow(tab)), function(x) {
    sd(mat[x,], na.rm = T)/mean(as.numeric(mat[x,]), na.rm = T)
  })
  # Keep only the iRTs with a CV < 50%:
  mat <- mat[cv < 0.5,]
  # I take as a reference the median of the runs (iRT signal) in the 4th biological repeat:
  # This is to keep the values close to the ones of the NS in the R4, where there was an issue with the spike and I can't perform normalisation
  med <- sapply(seq_len(ncol(mat)), function(x) {
    median(mat[,x], na.rm = T)
  })
  if (grepl("Intensity.ProFI_S4_R4", colnames(mat)[1])) {
    ref <- median(med, na.rm = T)
    print(ref)
  }
  names(med) <- colnames(mat)
  iRT[[i]] <- med
}

# Median of the normalisation factors:
for (i in seq_along(iRT)) {
  veci <- iRT[[i]]
  iRT[[i]] <- veci / ref
}
```

**The signal is very low for the iRTs in the TiO2_R4 unstimulated. I decide to not normalise these runs to avoid creating a bias with very low normalisation factors...**

```{r}
iRT$TiO2_R4[names(iRT$TiO2_R4) == "Intensity.ProFI_S4_R4_CCF01414_E.TiO2_NS.Inj1"] <- 1
iRT$TiO2_R4[names(iRT$TiO2_R4) == "Intensity.ProFI_S4_R4_CCF01414_E.TiO2_NS.Inj2"] <- 1
iRT$TiO2_R4[names(iRT$TiO2_R4) == "Intensity.ProFI_S4_R4_CCF01414_E.TiO2_NS.Inj3"] <- 1
```

```{r}
lf <- lf[order(names(lf))]
iRT <- iRT[order(names(iRT))]
for (i in seq_along(lf)) {
  tab <- lf[[i]]
  mat <- tab[,grepl("Intensity.", names(tab), fixed = T) & grepl("_Parsed", names(tab), fixed = T)]
  mat[mat == 0] <- NA
  mat <- apply(mat, 2, as.numeric)
  norm <- iRT[[i]]
  for (j in seq_len(ncol(mat))) {
    mat[,j] <- mat[,j]/norm[names(norm) == gsub("_Parsed", "", colnames(mat)[j])]
  }
  colnames(mat) <- gsub("_Parsed", "_iRTNorm", colnames(mat))
  tab <- cbind(tab, mat)
  lf[[i]] <- tab
}
rm(iRT)
```


# Data cleaning

For the replicate 1 (R1) of the TiO2, we noticed that the injection 1 were not ran in optimum conditions. There are more than 50% missing values in these runs. We remove them.

I also remove the R1_S120.Inj3bis that has a lower intensity than the others, which would impact the statistical analysis. And the RR_ProFI_S4_R4_CCF01414_EpTyr_NS.Inj3.

```{r, echo = F}
lf$TiO2_R1 <- lf$TiO2_R1[,!grepl(".Inj1_iRTNorm", names(lf$TiO2_R1))]
lf$TiO2_R1 <- lf$TiO2_R1[,!grepl("S120.Inj3bis_iRTNorm", names(lf$TiO2_R1))]
lf$pYIP_R4 <- lf$pYIP_R4[,!grepl("RR_ProFI_S4_R4_CCF01414_EpTyr_NS.Inj3_iRTNorm", names(lf$pYIP_R4))]
```



--------------------------------------------------------------------------------

# Number of sites

```{r}
n <- length(unique(names(psiteInfo)))
vec <- vector(mode = "character")
for (el in psiteInfo) {
  vec <- c(vec, el$MonoSites)
}
vec <- unique(vec)
```

There are `r n` unique sites or combination of sites (from multiply phosphorylated peptides) in the whole data set. These correspond to `r length(vec)` individual phosphorylation sites.

```{r}
par(mar = c(1, 1, 4, 1))
npprot <- sapply(lf, function(x) length(unique(x$Accession)))
npsites <- sapply(lf, function(x) length(unique(x$phosphoSite)))
lvenn <- lapply(lf, function(x) unique(x$phosphoSite))
print("Number of unique site in the analysis:")
print(length(unique(unlist(lvenn))))
# Draw venn diagrams:
## pYIP:
l <- lvenn[grepl("pYIP", names(lvenn))]
print("Number of unique site in the pYIP:")
print(length(unique(unlist(l))))
vec <-  sapply(l, length)
intAB <- length(intersect(l[[1]], l[[2]]))
intBC <- length(intersect(l[[3]], l[[2]]))
intAC <- length(intersect(l[[3]], l[[1]]))
intABC <- length(intersect(intersect(l[[3]], l[[2]]), l[[1]]))
v1 <- venneuler(c(A = as.numeric(vec[1]), B = as.numeric(vec[2]), C = as.numeric(vec[3]), "A&B" = intAB, "B&C" = intBC, "A&C" = intAC, "A&B&C" = intABC))

plot(v1)
title("pYIP\nRep 1, 4, 5")
venn(l)

## TiO2:
l <- lvenn[grepl("TiO2", names(lvenn))]
print("Number of unique site in the TiO2:")
print(length(unique(unlist(l))))
vec <-  sapply(l, length)
intAB <- length(intersect(l[[1]], l[[2]]))
intBC <- length(intersect(l[[3]], l[[2]]))
intAC <- length(intersect(l[[3]], l[[1]]))
intAD <- length(intersect(l[[4]], l[[1]]))
intBD <- length(intersect(l[[4]], l[[2]]))
intCD <- length(intersect(l[[4]], l[[3]]))
intABC <- length(intersect(intersect(l[[3]], l[[2]]), l[[1]]))
intABD <- length(intersect(intersect(l[[4]], l[[2]]), l[[1]]))
intACD <- length(intersect(intersect(l[[3]], l[[4]]), l[[1]]))
intABCD <- length(intersect(intersect(intersect(l[[3]], l[[4]]), l[[1]]), l[[2]]))
v2 <- venneuler(c(A = as.numeric(vec[1]), B = as.numeric(vec[2]), C = as.numeric(vec[3]), D = as.numeric(vec[4]), "A&B" = intAB, "B&C" = intBC, "A&C" = intAC, "A&D" = intAD, "D&C" = intCD, "D&B" = intBD, "A&B&C" = intABC, "A&B&D" = intABD, "A&C&D" = intACD, "A&B&C&D" = intABCD))

plot(v2)
title("TiO2\nRep 1, 3, 4, 5")

venn(l)

pdf("Figures/Venns.pdf", useDingbats=FALSE, 11.69, 8.27)
plot(v1)
title("pYIP\nRep 1, 4, 5")
plot(v2)
title("TiO2\nRep 1, 3, 4, 5")
dev.off()
```

## For figure 1

From now on, I will differenciate phosphorylation sites that have a localisation score >75% (Class 1), >0.5 and under or equal to 0.75 (Class 2) and >0.25 and under or equal to 0.5 (Class 3).

```{r}
for (i in 1:length(lf)) {
        tab <- lf[[i]]
        loc <- as.numeric(sub("%","",tab$Localization.prob))
        vec <- ifelse(loc > 0.75, "Class 1", "LowLocScore")
        vec[loc>0.5 & loc<=0.75] <- "Class 2"
        vec[loc>0.25 & loc<=0.5] <- "Class 3"
        tab$LocalisationClass <-  vec
        lf[[i]] <- tab
}
``` 

The multiply-phosphorylated sites do not get a score. For sites identified multiple times, I keep the highest localisation score for the pie chart.

```{r}
lclasses <- unlist(sapply(lf, function(x) x$LocalisationClass))
lpsites <- unlist(sapply(lf, function(x) x$phosphoSites))
tabClasses <- cbind(lpsites, lclasses)
tabClasses <- tabClasses[order(tabClasses[,2]),]
tabCpY <- tabClasses[grepl("pYIP", row.names(tabClasses)),]
tabCpY <- tabCpY[!duplicated(tabCpY[,1]),]
tabCTiO2 <- tabClasses[grepl("TiO2", row.names(tabClasses)),]
tabCTiO2 <- tabCTiO2[!duplicated(tabCTiO2[,1]),]

ClassespY <- table(tabCpY[,2])
ClassesTiO2 <- table(tabCTiO2[,2])
gtab <- rbind(melt(ClassesTiO2), melt(ClassespY))
gtab$Sample <- rep(c("TiO2", "pYIP"), each = 4)
gtab <- gtab[order(gtab$Var1),]

g <- ggplot(gtab, aes(x = "", y = value, fill = Var1))+ geom_bar(width = 1, stat = "identity", position = "fill") + coord_polar("y") + xlab("Biological replicate") + ylab("Proportion") + ggtitle("Class of the identified phosphorylation sites") + scale_fill_manual(values = mlpcolours[c(1:4)]) + theme_minimal() + facet_wrap(~Sample)  
print(g)

kable(gtab)

pdf("Figures/ClassPhospho.pdf", useDingbats=FALSE, 11.69, 8.27)
print(g)
dev.off()
```

For the monophosphorylations, I keep the phosphosites Class 1 and Class 2.

```{r}
for (i in 1:length(lf)) {
        tab1 <- lf[[i]]
        tab1 <- tab1[tab1$LocalisationClass == "Class 1" | tab1$LocalisationClass == "Class 2" | grepl("+", tab1$phosphoSites, fixed = T),]
        lf[[i]] <- tab1
}
n <- length(
  unique(
    unlist(
      sapply(lf, function(x) {
        x$phosphoSites
      })
    )
  )
)
```

After this filter, there are `r n` unique phosphosites.

Number of phosphorylated amino-acids identified in the study.

```{r}
# Number of amino acids:
################################################################################
laa <- lapply(lf, function(x) {
  lapply(x$phosphoSites, function(y) {
    psiteInfo[names(psiteInfo) == y][[1]]$MonoSites
  })
})
laaID <- lapply(laa, function(x) {
  unique(unlist(x))
})
  
uniqueAAID <- sapply(psiteInfo, function(x) {
  x$MonoSites
})
uniqueAAID <- unique(unlist(uniqueAAID))
numAATot <- as.character(
  sapply(uniqueAAID, function(y) {
    gsub("[1234567890]", "", strsplit(y, "_", fixed = T)[[1]][2])
  })
)
numAATot <- gsub(";", "", numAATot, fixed = T)
numAATot <- table(numAATot)

uniqueProtID <- sapply(names(psiteInfo), function(x) {
  strsplit(x, "_")[[1]][1]
})
uniqueProtID <- unique(uniqueProtID)
```

Total number of pS, pT, pY identified: `r numAATot`. These are present on `r length(uniqueProtID)` proteins.

```{r}
laa <- lapply(laaID, function(x) {
  as.character(
    sapply(x, function(y) {
      gsub("[1234567890]", "", strsplit(y, "_", fixed = T)[[1]][2])
    })
  )
})
laa <- sapply(laa, function(x) {
  gsub(";", "", x, fixed = T)
})
aa <- sapply(laa, table)
name <- sapply(laa, function(x) names(table(x)))

aminoacidtable <- aa


# Cumulative:
## pYIP:
aatot <- unique(c(laaID[[1]], laaID[[2]]))
vec <- sapply(aatot, function(x) {
  if (x %in% names(psiteInfo)) {
    psiteInfo[names(psiteInfo) == x][[1]]$MonoSites
  } else {
    NA
  }
})
vec <- unique(vec)
vec <- sapply(vec, function(x) {
  gsub("[1234567890]", "", strsplit(x, "_", fixed = T)[[1]][2])
})
vec <- sapply(vec, function(x) {
  gsub(";", "", x, fixed = T)
})
aa2 <- cbind("pYIP_R1" = aa[,1], "pYIP_R3"=table(vec))

aatot <- unique(c(laaID[[1]], laaID[[2]], laaID[[3]]))
vec <- sapply(aatot, function(x) {
  psiteInfo[names(psiteInfo) == x][[1]]$MonoSites
})
vec <- unique(vec)
vec <- sapply(vec, function(x) {
  gsub("[1234567890]", "", strsplit(x, "_", fixed = T)[[1]][2])
})
vec <- sapply(vec, function(x) {
  gsub(";", "", x, fixed = T)
})
aa2 <- cbind(aa2, "pYIP_R5"=table(vec))

## TiO2:
aatot <- unique(c(laaID[[4]], laaID[[5]]))
vec <- sapply(aatot, function(x) {
  psiteInfo[names(psiteInfo) == x][[1]]$MonoSites
})
vec <- unique(vec)
vec <- sapply(vec, function(x) {
  gsub("[1234567890]", "", strsplit(x, "_", fixed = T)[[1]][2])
})
vec <- sapply(vec, function(x) {
  gsub(";", "", x, fixed = T)
})
aa1 <- cbind("TiO2_R1" = aa[,4], "TiO2_R3"=table(vec))

aatot <- unique(c(laaID[[4]], laaID[[5]], laaID[[6]]))
vec <- sapply(aatot, function(x) {
  psiteInfo[names(psiteInfo) == x][[1]]$MonoSites
})
vec <- unique(vec)
vec <- sapply(vec, function(x) {
  gsub("[1234567890]", "", strsplit(x, "_", fixed = T)[[1]][2])
})
vec <- sapply(vec, function(x) {
  gsub(";", "", x, fixed = T)
})
aa1 <- cbind(aa1, "TiO2_R4"=table(vec))

aatot <- unique(c(laaID[[4]], laaID[[5]], laaID[[6]], laaID[[7]]))
vec <- sapply(aatot, function(x) {
  psiteInfo[names(psiteInfo) == x][[1]]$MonoSites
})
vec <- unique(vec)
vec <- sapply(vec, function(x) {
  gsub("[1234567890]", "", strsplit(x, "_", fixed = T)[[1]][2])
})
vec <- sapply(vec, function(x) {
  gsub(";", "", x, fixed = T)
})
aa1 <- cbind(aa1, "TiO2_R5"=table(vec))

laatables <- list("TiO2" = aa1, "pYIP"= aa2)

for (i in seq_along(laatables)) {
  tab <- laatables[[i]]
  tab <- rbind(tab, colSums(tab))
  row.names(tab)[nrow(tab)] <- "total"
  laatables[[i]] <- tab
}

print("Number of identified phosphorylated amino acid:")
print(laatables)
################################################################################

# Not referenced in PhosphoSitePlus:
laa <- lapply(lf, function(x) {
  x$phosphoSites[x$PSPMouse == ""]
})

laaID <- lapply(laa, function(x) {
  unique(unlist(x))
})
  
laa <- lapply(laaID, function(x) {
  as.character(
    sapply(x, function(y) {
      gsub("[1234567890]", "", strsplit(y, "_", fixed = T)[[1]][2])
    })
  )
})
laa <- sapply(laa, function(x) {
  gsub(";", "", x, fixed = T)
})
laa <- sapply(laa, function(x) {
  unlist(strsplit(x, "+", fixed = T))
})
aa <- sapply(laa, table)
name <- sapply(laa, function(x) names(table(x)))

aa <- rbind(aa, "total" = colSums(aa))

# Cumulative:
## pYIP:
aatot <- unique(c(laaID[[1]], laaID[[2]]))
vec <- sapply(aatot, function(x) {
  psiteInfo[names(psiteInfo) == x][[1]]$MonoSites
})
vec <- unlist(vec)
vec <- unique(vec)
vec <- sapply(vec, function(x) {
  gsub("[1234567890]", "", strsplit(x, "_", fixed = T)[[1]][2])
})
vec <- sapply(vec, function(x) {
  gsub(";", "", x, fixed = T)
})
vec <- sapply(vec, function(x) {
  unlist(strsplit(x, "+", fixed = T))
})
aa2 <- cbind("pYIP_R1" = aa[1:3,1], "pYIP_R4"=table(vec))

aatot <- unique(c(laaID[[1]], laaID[[2]], laaID[[3]]))
vec <- sapply(aatot, function(x) {
  psiteInfo[names(psiteInfo) == x][[1]]$MonoSites
})
vec <- unlist(vec)
vec <- unique(vec)
vec <- sapply(vec, function(x) {
  gsub("[1234567890]", "", strsplit(x, "_", fixed = T)[[1]][2])
})
vec <- sapply(vec, function(x) {
  gsub(";", "", x, fixed = T)
})
vec <- sapply(vec, function(x) {
  unlist(strsplit(x, "+", fixed = T))
})
aa2 <- cbind(aa2, "pYIP_R5"=table(vec))

## TiO2:
aatot <- unique(c(laaID[[4]], laaID[[5]]))
vec <- sapply(aatot, function(x) {
  psiteInfo[names(psiteInfo) == x][[1]]$MonoSites
})
vec <- unlist(vec)
vec <- unique(vec)
vec <- sapply(vec, function(x) {
  gsub("[1234567890]", "", strsplit(x, "_", fixed = T)[[1]][2])
})
vec <- sapply(vec, function(x) {
  gsub(";", "", x, fixed = T)
})
vec <- sapply(vec, function(x) {
  unlist(strsplit(x, "+", fixed = T))
})
aa1 <- cbind("TiO2_R1" = aa[1:3,4], "TiO2_R3"=table(vec))

aatot <- unique(c(laaID[[4]], laaID[[5]], laaID[[6]]))
vec <- sapply(aatot, function(x) {
  psiteInfo[names(psiteInfo) == x][[1]]$MonoSites
})
vec <- unlist(vec)
vec <- unique(vec)
vec <- sapply(vec, function(x) {
  gsub("[1234567890]", "", strsplit(x, "_", fixed = T)[[1]][2])
})
vec <- sapply(vec, function(x) {
  gsub(";", "", x, fixed = T)
})
vec <- sapply(vec, function(x) {
  unlist(strsplit(x, "+", fixed = T))
})
aa1 <- cbind(aa1, "TiO2_R4"=table(vec))

aatot <- unique(c(laaID[[4]], laaID[[5]], laaID[[6]], laaID[[7]]))
vec <- sapply(aatot, function(x) {
  psiteInfo[names(psiteInfo) == x][[1]]$MonoSites
})
vec <- unlist(vec)
vec <- unique(vec)
vec <- sapply(vec, function(x) {
  gsub("[1234567890]", "", strsplit(x, "_", fixed = T)[[1]][2])
})
vec <- sapply(vec, function(x) {
  gsub(";", "", x, fixed = T)
})
vec <- sapply(vec, function(x) {
  unlist(strsplit(x, "+", fixed = T))
})
aa1 <- cbind(aa1, "TiO2_R5"=table(vec))

laatablesNotRef <- list("TiO2" = aa1, "pYIP"= aa2)

print("Number of identified phosphorylated amino acid that are not reported in the mouse PhosphoSitePlus data:")
print(laatablesNotRef)

aatot <- unique(c(laaID[[7]], laaID[[1]], laaID[[2]], laaID[[3]], laaID[[4]], laaID[[5]], laaID[[6]]))
vec <- sapply(aatot, function(x) {
  psiteInfo[names(psiteInfo) == x][[1]]$MonoSites
})
vec <- unlist(vec)
vec <- unique(vec)
vec <- sapply(vec, function(x) {
  gsub("[1234567890]", "", strsplit(x, "_", fixed = T)[[1]][2])
})
vec <- sapply(vec, function(x) {
  gsub(";", "", x, fixed = T)
})
vec <- sapply(vec, function(x) {
  unlist(strsplit(x, "+", fixed = T))
})

numtotSnotref <- length(vec[vec=="S"])
numtotTnotref <- length(vec[vec=="T"])
numtotYnotref <- length(vec[vec=="Y"])
```

Number of each amino-acid in the table:

```{r}
# gtab <- melt(aminoacidtable)
gtab <- rbind(data.frame("Var1"=row.names(laatables[[1]]), "value"=laatables[[1]][,4], "Sample" = rep("TiO2", 4)), data.frame("Var1"=row.names(laatables[[2]]), "value"=laatables[[2]][,3], "Sample" = rep("pYIP", 4)))
gtab <- gtab[order(gtab$Var1),]
# gtab$Sample <- sapply(as.character(gtab$Var2), function(x) {
#   strsplit(x, "_", fixed = T)[[1]][1]
# })
gtab <- gtab[gtab$Var1 != "total",]
g <- ggplot(gtab, aes(x = "", y = value, fill = Var1)) + geom_bar(width = 1, stat = "identity", position = "fill") + coord_polar("y") + xlab("Biological replicate") + ylab("Proportion") + ggtitle("Amino-acid composition of data sets") + scale_fill_manual(values = mlpcolours[4:6]) + theme_minimal() + facet_wrap(.~Sample)
print(g)

aa <- cbind(laatables[[1]], laatables[[2]])

gtab <- melt(aa)
gtab$Var2 <- as.character(gtab$Var2)
gtab$Var2[gtab$Var2 == "pYIP_R3"] <- "pYIP_R4"
gtab$value[gtab$Var1 == "total" & (gtab$Var2 == "pYIP_R1" | gtab$Var2 == "pYIP_R4")] <- NA
gtab$value[gtab$Var1 == "total" & (gtab$Var2 == "TiO2_R1" | gtab$Var2 == "TiO2_R3" | gtab$Var2 == "TiO2_R4")] <- NA
gtab <- gtab[!is.na(gtab$value),]
gtab <- gtab[order(gtab$Var1),]
gtab$Var2 <- as.character(gtab$Var2)
gtab$Sample <- sapply(as.character(gtab$Var2), function(x) {
  strsplit(x, "_", fixed = T)[[1]][1]
})
gtab$Var2[gtab$Var1 == "total"] <- gsub("pYIP_", "", gtab$Var2[gtab$Var1 == "total"], fixed = T)
gtab$Var2[gtab$Var1 == "total"] <- gsub("TiO2_", "", gtab$Var2[gtab$Var1 == "total"], fixed = T)

gtab$Var1 <- factor(as.character(gtab$Var1), levels = sort(as.character(unique(gtab$Var1)))[c(3, 1, 2, 4)])
g2 <- ggplot(gtab, aes(x = Var1, y = value, fill = Var1, group = Var2)) + geom_bar(stat = "identity", position = "dodge", col = "black")  + xlab("Phosphorylated amino-acid") + ylab("Cumulative number of identified and quantified phosphorylated site") + ggtitle("Amino-acid composition of data sets") + scale_fill_manual(values = c("grey36", mlpcolours[4:6])) + theme_minimal() + facet_wrap(.~Sample, scale = "free_y")
print(g2)

pdf("Figures/aa_Total.pdf", useDingbats=FALSE, 11.69, 8.27)
print(g)
print(g2)
dev.off()

# Not referenced in psitePlus:
###############################
aa <- cbind(laatablesNotRef[[1]], laatablesNotRef[[2]])
aa <- rbind(aa, "total" = colSums(aa))

gtab <- melt(aa)
gtab$value[gtab$Var1 == "total" & (gtab$Var2 == "pYIP_R1" | gtab$Var2 == "pYIP_R4")] <- NA
gtab$value[gtab$Var1 == "total" & (gtab$Var2 == "TiO2_R1" | gtab$Var2 == "TiO2_R3" | gtab$Var2 == "TiO2_R4")] <- NA
gtab <- gtab[!is.na(gtab$value),]
gtab <- gtab[order(gtab$Var1),]
gtab$Var2 <- as.character(gtab$Var2)
gtab$Sample <- sapply(as.character(gtab$Var2), function(x) {
  strsplit(x, "_", fixed = T)[[1]][1]
})
gtab$Var2[gtab$Var1 == "total"] <- gsub("pYIP_", "", gtab$Var2[gtab$Var1 == "total"], fixed = T)
gtab$Var2[gtab$Var1 == "total"] <- gsub("TiO2_", "", gtab$Var2[gtab$Var1 == "total"], fixed = T)

gtab$Var1 <- factor(as.character(gtab$Var1), levels = sort(as.character(unique(gtab$Var1)))[c(3, 1, 2, 4)])
g2 <- ggplot(gtab, aes(x = Var1, y = value, fill = Var1, group = Var2)) + geom_bar(stat = "identity", position = "dodge", col = "black")  + xlab("Phosphorylated amino-acid") + ylab("Cumulative number of identified and quantified phosphorylated site") + ggtitle("Phosphorylation sites not previously referenced in phosphositeplus") + scale_fill_manual(values = c("grey36", mlpcolours[4:6])) + theme_minimal() + facet_wrap(.~Sample, scale = "free_y")
print(g2)

pdf("Figures/aa_NotRef.pdf", useDingbats=FALSE, 11.69, 8.27)
print(g2)
dev.off()
```

*I think that the high number of S/T that we find in the pYIP can be explained by the contiguous sites in multiply-phosphorylated peptides.*


--------------------------------------------------------------------------------

# Data merge and normalisation

For the monophosphorylations, I keep the phosphosites Class 1 and Class 2 before merging the tables.
I log2-transform the data set.

```{r}
lmat <- list()
for (i in 1:length(lf)) {
        tab1 <- lf[[i]]
        # tab1 <- tab1[tab1$LocalisationClass == "Class 1" | tab1$LocalisationClass == "Class 2" | grepl("+", tab1$phosphoSites, fixed = T),]
        mat11 <- tab1[,grepl("iRTNorm", names(tab1), fixed = T)]
        names(mat11) <- gsub("Intensity.", "", names(mat11), fixed = T)
        n <- sapply(1:nrow(mat11), function(x) length(mat11[x,][is.na(mat11[x,])]))
        tab1 <- tab1[n<ncol(mat11),]
        mat11 <- mat11[n<ncol(mat11),] # Remove the lines with only missing values
        row.names(mat11) <- tab1$phosphoSites
        lmat[[length(lmat)+1]] <- log2(mat11)
}
names(lmat) <- names(lf)

par(mar = c(3, 2, 3, 1), mfrow = c(1,1))
for (i in 1:length(lmat)) {
  matl <- lmat[[i]]
  title <- paste0(names(lmat[i]), " log2 intensity values for all psite\nin the injection replicates")
  na <- dimnames(matl)[[2]]
  na <- sapply(na, function(x) strsplit(x, "_"))
  na <- sapply(na, function(x) x[grepl("Inj", x)])
  dimnames(matl)[[2]] <- na
  boxplot(matl, main = title, las = 2)
}
```

Pairwise plot with log2 values observed per condition per biological replicate:

```{r}
for (i in seq_along(lmat)) {
  mat <- lmat[[i]]
   for (el2 in c("NS", "S15.", "S30.", "S120.", "S300", "S600.")) {
    sub <- mat[,grepl(el2, dimnames(mat)[[2]], fixed = T)]
    tit <- paste0(names(lmat)[i], " ", el2)
    pairs(log(sub, 2), main = tit, lower.panel = panel.smooth, diag.panel = panel.hist, upper.panel = panel.cor)
   }
}
dev.off()
```

## QC figures

I make a figure with correlation between all runs for the supplementary data of the paper:

```{r}
par(mfrow = c(1,1), mar = c(1,1,1,1))
col1 <- colorRampPalette(c("darkred", "red", "indianred1", "white", "cornflowerblue", "blue", "darkblue", "black")) 
cormat <- merge(data.frame("psiteID"= row.names(lmat[[1]]), lmat[[1]]), data.frame("psiteID"= row.names(lmat[[2]]), lmat[[2]]), by = "psiteID", all = T)
for (i in 3:length(lmat)) {
  cormat <- merge(cormat, data.frame("psiteID"= row.names(lmat[[i]]), lmat[[i]]), by = "psiteID", all = T)
}
rn <- as.character(cormat$psiteID)
cormat <- as.matrix(cormat[,2:ncol(cormat)])
row.names(cormat) <- rn
cormat <- log10(cormat)
M <- cor(cormat, use = "pairwise.complete.obs")

corrplot(M, method = "shade", col=col1(600), mar = c(1,1,1,1), tl.cex = 0.25, tl.col = "black", order = "hclust")  

# pYIP:
cormat1 <- cormat[,grepl("pTyr", colnames(cormat))]
M <- cor(cormat1, use = "pairwise.complete.obs")
corrplot(M, method = "shade", col=col1(600), mar = c(1,1,1,1), tl.cex = 0.25, tl.col = "black", order = "hclust")  
vec <- sapply(colnames(M), function(x) {
  strsplit(x, "Inj", fixed = T)[[1]][1]
})
M1 <- M
for (i in seq_len(nrow(M))) {
  for (j in seq_len(ncol(M))) {
    if (vec[j] != vec[i]) {
      M[i,j] <- NA
    }
    if (colnames(M)[j] == row.names(M)[i]) {
      M[i,j] <- NA
    }
  }
}
gtab <- melt(M)
ggplot(gtab, aes(x = value)) + geom_histogram(bins = 50) + theme_bw() + ggtitle("Correlation values between runs in the pYIP data set\nbefore data analysis (R from cor() function)")

pdf("Figures/QCTechnicalReplicates_pYIP.pdf", useDingbats=FALSE, 11.69, 8.27) # width and height in inches.
corrplot(M1, method = "shade", col=col1(600), mar = c(1,1,1,1), tl.cex = 0.25, tl.col = "black", order = "hclust")  
ggplot(gtab, aes(x = value)) + geom_histogram(bins = 50) + theme_bw() + ggtitle("Correlation values between runs in the pYIP data set\nbefore data analysis (R from cor() function)")
dev.off()

# TiO2:
cormat1 <- cormat[,grepl("TiO2", colnames(cormat))]
M <- cor(cormat1, use = "pairwise.complete.obs")
corrplot(M, method = "shade", col=col1(600), mar = c(1,1,1,1), tl.cex = 0.25, tl.col = "black", order = "hclust")
vec <- sapply(colnames(M), function(x) {
  strsplit(x, "Inj", fixed = T)[[1]][1]
})
M1 <- M
for (i in seq_len(nrow(M))) {
  for (j in seq_len(ncol(M))) {
    if (vec[j] != vec[i]) {
      M[i,j] <- NA
    }
    if (colnames(M)[j] == row.names(M)[i]) {
      M[i,j] <- NA
    }
  }
}
gtab <- melt(M)
ggplot(gtab, aes(x = value)) + geom_histogram(bins = 50) + theme_bw() + ggtitle("Correlation values between runs in the TiO2 data set\nbefore data analysis (R from cor() function)")

pdf("Figures/QCTechnicalReplicates_TiO2.pdf", useDingbats=FALSE, 11.69, 8.27) # width and height in inches.
corrplot(M1, method = "shade", col=col1(600), mar = c(1,1,1,1), tl.cex = 0.25, tl.col = "black", order = "hclust")  
ggplot(gtab, aes(x = value)) + geom_histogram(bins = 50) + theme_bw() + ggtitle("Correlation values between runs in the TiO2 data set\nbefore data analysis (R from cor() function)")
dev.off()
```

```{r}
save(lf, file = "RData/05_ParsedTablesNormPSP.RData")
```

--------------------------------------------------------------------------------

```{r}
sessionInfo()
```