---
title: "Mapping phosphorylation sites to known sites in the databases"
author: "Marie Locard-Paulet"
date: '`r Sys.Date()`'
output: html_document
---

```{r, echo=F, message=F}
knitr::opts_chunk$set(echo = FALSE, warning = F)

require(knitr)
require(Biostrings)
require(ggplot2)
require(reshape2)
```


```{r}
load("RData/02_psiteInfo.RData")
```

This document retrieves the information from specific phosphorylation sites databases in order to analyse the label-free MS data sets of phospho-enriched peptides from primary T-cells.

I use the mapping files from [PhosphOrtholog](http://www.phosphortholog.com/), Chadhuri et al. 2015 (10.1186/s12864-015-1820-x). From these, I retrieve the Uniprot IDs of Rat and Human corresponding to the ones I have in my data set.

```{r, eval = F}
MappingHM <- read.table("T:/DB/Mapping/PhosphOrtholog_20170623/Mapping_H_M.csv", sep = " ", header = T)
MappingRM <- read.table("T:/DB/Mapping/PhosphOrtholog_20170623/Mapping_R_M.csv", sep = " ", header = T)
MappingM <- rbind(MappingRM[,c(2,1)], MappingHM[,c(2,1)]) # The mouse ID is in the first column
rm(MappingHM, MappingRM)

# Find the ones I need:
myAcc <- names(psiteInfo)
myAcc <- sapply(myAcc, function(x) {
  strsplit(x, "_", fixed = T)[[1]][1]
})
myAcc <- sapply(myAcc, function(x) {
  strsplit(x, ";", fixed = T)[[1]]
})
myAcc <- unique(unlist(myAcc))

MappingM <- MappingM[MappingM$UniProt.SwissProt.Accession.2 %in% myAcc,]
MappingList <- lapply(as.character(unique(MappingM[,1])), function(x){
  as.character(MappingM[MappingM==x,2])
})
names(MappingList) <- as.character(unique(MappingM[,1]))

save(list = c("MappingList", "MappingM", "myAcc"), file = "RData/DBMapping/MappingPhospho_01.Rdata") # the names are the mouse ID
```

```{r}
load("RData/DBMapping/MappingPhospho_01.Rdata")
```

Protein accessions that are in my data but not in my mapping file:

```{r, echo = F}
myAcc[!(myAcc %in% names(MappingList))]
```

# Retrieve information from PhosphositePlus

I work with the tables downloaded from the [PhosphositePlus website](http://www.phosphosite.org/homeAction.action) the 17th of August 2018:

* `PhosphoSitePlus_20180817/Kinase_Substrate_Dataset` contains the kinase/substrate relationships.
* `PhosphoSitePlus_20180817/Phosphorylation_site_dataset` contains the informations regarding the phosphosites, including their position in the sequence.

I merge the tables based on the substrate sequence window (in upper cases). In order to map kinases and domains to our result table, I use the sequence window to map the phosphosites we identify to the phosphositeplus table.

On top of keeping the kinase/substrate and domains information, I create a column called  `InPsitePlus` that contains "yes" or "no" depending on if this phosphosite has already been references in the database.

```{r, eval = F}
# ltabK <- list()
# ltabK[[1]] <- read.table("PhosphoSitePlus_20180817/Kinase_Substrate_Dataset", as.is = T, comment.char = "#", sep = "\t", header = T, skip = 3)
# ltabK[[2]] <- read.csv("PhosphoSitePlus_20180817/Phosphorylation_site_dataset", as.is = T, comment.char = "#", sep = "\t", header = T, skip = 3)
# ltabK[[1]]$SITE_...7_AA <- toupper(ltabK[[1]]$SITE_...7_AA)
# ltabK[[2]]$SITE_...7_AA <- toupper(ltabK[[2]]$SITE_...7_AA)
# tabK <- merge(ltabK[[1]], ltabK[[2]], by = "SITE_...7_AA", all = T)
# SITE_GRP_ID <- as.character(tabK$SITE_GRP_ID.x)
# SITE_GRP_ID[is.na(SITE_GRP_ID)] <- as.character(tabK$SITE_GRP_ID.y)[is.na(SITE_GRP_ID)]
# tabK$ALL_SITEGPID <- SITE_GRP_ID
# save(tabK, file = "RData/DBMapping/PsitePlus.RData")
```

```{r, eval = F}
ltabK <- list()
ltabK[[1]] <- read.table("PhosphoSitePlus_20180817/Kinase_Substrate_Dataset", as.is = T, comment.char = "#", sep = "\t", header = T, skip = 3)
ltabK[[2]] <- read.csv("PhosphoSitePlus_20180817/Phosphorylation_site_dataset", as.is = T, comment.char = "#", sep = "\t", header = T, skip = 3)
ltabK[[1]]$SITE_...7_AA <- toupper(ltabK[[1]]$SITE_...7_AA)
ltabK[[2]]$SITE_...7_AA <- toupper(ltabK[[2]]$SITE_...7_AA)
names(ltabK) <- c("KinaseSubstrate", "PhosphoSite")

save(ltabK, file = "RData/DBMapping/PsitePlus.RData")
```

```{r}
load(file = "RData/DBMapping/PsitePlus.RData")
```

I find the sequence windows of the data set, and reduce them from 15- to 7-aa length. I split the IDs with protein ambiguity ("protA;protB") to be sure not to miss them.

```{r}
load("RData/01_ParsedTables.RData")

seq <- sapply(lf, function(x) {
  cbind(x$Sequence.window, x$phosphoSite)
})

source("RFunctions/RBindList.R")

seq <- RBindList(seq)
seq <- seq[!duplicated(seq),]
# Split rows with multiple protein IDs:
temp <- seq[grepl(";", seq[,1]),]
vec1 <- lapply(temp[,1], function(x) {
  strsplit(x, ";", fixed = T)[[1]]
})
vec2 <- sapply(seq_len(nrow(temp)), function(x) {
  rep(temp[x,2],sapply(vec1, length)[x])
})
temp <- cbind(unlist(vec1), unlist(vec2))
temp <- temp[!duplicated(temp),]
seq <- rbind(seq[!grepl(";", seq[,1]),], temp)
colnames(seq) <- c("Window15", "phosphoSites")
# Reduce the sequence length:
vec <- as.character(sapply(seq[,1], substr, 9, 23))
seq <- cbind(seq, "Window7" = vec)

# Remove redundancy:
seq <- seq[!(duplicated(seq[,2:3])),]
seq <- seq[order(seq[,2]),]
# Make IDs for multiple protein IDs:
seq <- seq[,2:3]
temp <- seq[grepl(";", seq[,1]),]
vecprot <- lapply(temp[,1], function(x) {
  strsplit(x, "_", fixed = T)[[1]][1]
})
vecprot <- lapply(vecprot, function(x) {
  strsplit(x, ";", fixed = T)[[1]]
})
vec2 <- sapply(seq_len(nrow(temp)), function(x) {
  rep(temp[x,2],sapply(vecprot, length)[x])
})
vec3 <- sapply(seq_len(nrow(temp)), function(x) {
  rep(temp[x,1],sapply(vecprot, length)[x])
})
temp <- cbind(unlist(vec3), unlist(vec2), unlist(vecprot))
temp <- temp[!duplicated(temp),]
colnames(temp) <- c("phosphoSites",  "Window7", "ProteinAccession")
temp2 <- seq[!(grepl(";", seq[,1])),]
temp2 <- cbind(temp2, "ProteinAccession" = sapply(temp2[,1], function(x) {
  strsplit(x, "_", fixed = T)[[1]][1]
}))
seq <- rbind(temp2, temp)
```

First, I look for the sites in our data set that are present in the mouse database from PhosphositePlus.

```{r}
psiteID <- names(psiteInfo)

tabPSPMouse <- ltabK$PhosphoSite[ltabK$PhosphoSite$ORGANISM == "mouse",]

seq <- cbind(seq, "SiteID" = paste0(seq[,3], "|", seq[,2]))
tabPSPMouse$SiteID <- paste0(tabPSPMouse$ACC_ID, "|", tabPSPMouse$SITE_...7_AA)

Known <- seq[,1][seq[,4] %in% as.character(tabPSPMouse$SiteID)]

PPIDs <- tabPSPMouse$SITE_GRP_ID[match(seq[,4], as.character(tabPSPMouse$SiteID))]
seq <- data.frame(seq, "PhosphoPlusIDsMouse" = PPIDs)
mapped <- unique(seq$phosphoSites[!is.na(seq$PhosphoPlusIDs)])
notmapped <- unique(seq$phosphoSites[is.na(seq$PhosphoPlusIDs)])
seqMouse <- seq
```

There are `r length(mapped)` phosphorylation sites of our data set mapped to PhosphositePlus mouse IDs due to exact identical sequence. Using this simple matching approach, `r length(notmapped)` of the phosphorylation sites of the data set are not mapped to PhosphositePlus IDs. In total, we have `r length(unique(as.character(seq$phosphoSites)))` phosphorylation sites in our data set.

Then, I look for the sites in our data set that are present in the all the databases from PhosphositePlus.

In order to gain information from these sites, I look for similar sequences using the "pairwiseAlignment()" function. 

```{r, echo = F, eval = F}
# I need to do it in a different way, performing the pairwise alignment on the entire protein sequence in order to avoid mapping similar sequences at different parts of the protein.
# seq2 <- seq[seq$phosphoSites %in% as.character(notmapped),]
seq2 <- seq[!grepl("+", seq[,1], fixed = T),]
 
tabK <- ltabK$PhosphoSite[!(ltabK$PhosphoSite$ORGANISM == "mouse"),]

l <- list(); lOrga <- list()
for (i in 1:nrow(seq2)) {
        vari <- as.character(seq2$Window7[i])
        acc <- as.character(seq2$ProteinAccession[i])
        acc2 <- unlist(MappingList[names(MappingList)==acc])
        temp <- tabK[(tabK$ACC_ID %in% acc2),]
        temp$AllOrga <- temp$ORGANISM
        if (nrow(temp)>0) {
                vari2 <- gsub("_", "", temp$SITE_...7_AA)
                vari <- gsub("_", "", vari)
                matchList <- lapply(vari2, function(x){
                  pairwiseAlignment(AAString(vari), AAString(x), substitutionMatrix = "BLOSUM50", gapOpening = 10, gapExtension = 10, scoreOnly == FALSE)
                  })
                scores <- sapply(matchList, score)
                IDs <- unique(temp$SITE_GRP_ID[scores>58 & substr(vari, 8,8) == substr(vari2, 8, 8)])
                Species <- unique(temp$AllOrga[scores>58 & substr(vari, 8,8) == substr(vari2, 8, 8)])
        }
        if (!(class(IDs) == "character" & length(IDs) == 0)) {
          l[[length(l)+1]] <- IDs
          names(l)[[length(l)]] <- as.character(seq2$phosphoSites)[i]
          lOrga[[length(lOrga)+1]] <- Species
          names(lOrga)[[length(lOrga)]] <- as.character(seq2$phosphoSites)[i]
        }
}

save(list = c("seq", "seq2", "l", "lOrga"), file = "RData/DBMapping/MappingPhospho_02.Rdata")
```

```{r, echo = F}
load("RData/DBMapping/MappingPhospho_02.Rdata")
```

On the `r length(l)` phosphorylation sites of the query, `r length(l[sapply(l, length)>0])` have sequence matches.

```{r, echo = F}
n <- sapply(lOrga, function(x){max(as.numeric(as.character(table(x))))})
```


```{r, eval = F}
temp <- seq2
temp$PSPNonMouseIDs <- sapply(l[match(temp$phosphoSites, names(l))], paste, collapse = "|")
temp$PSPNonMouseIDsOrganisms <- sapply(lOrga[match(temp$phosphoSites, names(lOrga))], paste, collapse = "|")
temp$PSPNonMouseIDsOrganisms[temp$PSPNonMouseIDsOrganisms == ""] <- NA
temp$PSPNonMouseIDs[temp$PSPNonMouseIDs == ""] <- NA
seq <- temp
```

```{r, eval = F}
ManualInput <- read.table("RData/DBMapping/ManualMapping.txt", sep = "\t", header = T, stringsAsFactors = F)
```

<!-- I manually add these in our data sets. -->

```{r, eval = F}
for (i in seq_len(nrow(ManualInput))) {
  l[[length(l)+1]] <- ManualInput[i,2]
  names(l)[length(l)] <- ManualInput[i,1]
}
rm(seq2)

seq$PhosphoPlusIDs <- as.character(seq$PhosphoPlusIDs)
for (i in seq_along(l)) {
  seq$PhosphoPlusIDs[seq$phosphoSites == names(l)[i]] <- paste(unique(l[[i]]), collapse = "|")
}
```

```{r, eval = F}
IsMapped <- unique(seq$phosphoSites[!is.na(seq$PSPNonMouseIDs) | !is.na(seq$PhosphoPlusIDsMouse)])
NotMapped <- unique(seq$phosphoSites[!(!is.na(seq$PSPNonMouseIDs) | !is.na(seq$PhosphoPlusIDsMouse))])
```

I map back PhosphositePlus IDs and informations to annotated tables.

```{r, eval = F}
for (i in seq_along(psiteInfo)) {
  li <- psiteInfo[[i]]
  li$PhosphoSitePlusMouse <- seq$PhosphoPlusIDsMouse[match(li$MonoSites, seq$phosphoSites)][!is.na(seq$PhosphoPlusIDsMouse[match(li$MonoSites, seq$phosphoSites)])]
  li$PhosphoSitePlusNotMouse <- seq$PSPNonMouseIDs[match(li$MonoSites, seq$phosphoSites)][!is.na(seq$PSPNonMouseIDs[match(li$MonoSites, seq$phosphoSites)])]
  psiteInfo[[i]] <- li
}
  
save(psiteInfo, file = "RData/03_psiteInfoPhosphoSitePlus.RData")
```

```{r}
load(file = "RData/03_psiteInfoPhosphoSitePlus.RData")
```

```{r, eval = F}
for (i in seq_along(lf)) {
  tab <- lf[[i]]
  tab$PSPAllSPecies <- sapply(tab$phosphoSites, function(x) {
    paste(psiteInfo[[which(names(psiteInfo) == x)]]$PhosphoSitePlusNotMouse, collapse = "|")
  })
  tab$PSPMouse <- sapply(tab$phosphoSites, function(x) {
    paste(as.character(psiteInfo[[which(names(psiteInfo) == x)]]$PhosphoSitePlusMouse), collapse = "|")
  })
  tab$PSPAllSPecies[is.na(tab$PSPAllSPecies) | tab$PSPAllSPecies == ""] <- tab$PSPMouse[is.na(tab$PSPAllSPecies) | tab$PSPAllSPecies == ""]
  lf[[i]] <- tab
}
  
save(lf, file = "RData/04_ParsedTablesPSP.RData")
```

```{r}
load(file = "RData/04_ParsedTablesPSP.RData")
```

```{r}
notmapped <- unique(unlist(sapply(lf, function(x) {x$phosphoSites[x$PSPAllSPecies == ""]})))
mapped <- unique(unlist(sapply(lf, function(x) {x$phosphoSites[x$PSPAllSPecies != ""]})))

vec <- sapply(psiteInfo[names(psiteInfo) %in% notmapped], function(x) {x$GeneID})
vec <- as.character(vec)
vec <- vec[grepl("_Y", vec, fixed = T)]
```

There are `r length(mapped)` and `r length(notmapped)` phosphorylation sites in our data set that are reported in phosphositePlus and not reported, respectively.

 sites from the pY-IP are not referenced in PhosphoSitePlus:
`r sort(vec)`.

```{r}
sessionInfo()
```